### DISCUSSION TITLE
Question regarding custom index listing templates

### DISCUSSION THREAD

**Discussion by dweng0** [quarto-dev/quarto-cli#9567] [Q&A]
### Description

I'm following this [documentation](https://quarto.org/docs/websites/website-listings-custom.html#sorting-filtering-and-pagination) regarding the cust listings and implementing sorting and filtering. The problem I encounter is that when rendering the example code (with a modified class to just point to the name) 

```html
<ul class="list">
	<% for (const item of items) { %>
	  <li <%= metadataAttrs(item) %>>
		<a href="<%- item.href %>" class="listing-name"><%= item.name %></a><br/>
		<span class="listing-name"><%= item.name %><span>
		<span>Test display</span>
	  </li>
	<% } %>
	</ul>
```

and an updated index frontmatter set:

```qmd
---
title: Autonity dashboards
format: 
  html:
    css: _resources/css/index-style.css
listing:
  contents: dashboards
  template: _templates/index.ejs
  sort-ui: [title, date]
  filter-ui: true
  page-size: 10
---
```

I end up with the only the `Test display` showing for the four items but no values from the frontmatter of the given item.

Note that I can get the list to display with out the filtering, but as soon as i add the `metadataAttrs(item)` stuff, it fails to work. 

To sanity check that the data was available. I console logged the item object in the for loop:

```html
<ul class="list">
<% for (const item of items) { console.log(item); %>
  ... rest of code
<% } %>
</ul>
```

I'm unable to debug it further, is there more details on what `metadataAttrs` is doing under the hood that could be causing this?

![image](https://github.com/quarto-dev/quarto-cli/assets/5268627/a0a4bcc5-8e44-4b72-b757-d7c90c09b541)

**Comment by mcanouil** [comment]
Here you are requesting `name` but if you don't have `name: something` in your content, then there is nothing to be displayed.

Look at the contents described in the documentation: https://quarto.org/docs/websites/website-listings-custom.html#metadata-file-listings
